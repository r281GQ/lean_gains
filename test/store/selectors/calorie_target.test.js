import { fromJS, Map } from 'immutable';
import moment from 'moment';
import * as _ from 'lodash';

import selector from './../../../src/store/selectors/calorie_target';
import { latestMeasurements } from './../../../shared/test_constants';


// const formValues = fromJS({
//   trainingDay: 20,
//   bmrCalculationMethod: 'harris-benedict',
//   protein: 2,
//   restDay: -20,
//   calorieSplit: 'recomp',
//   activity: 1.2,
//   restFatGrams: 0,
//   bodyFat: 15.3,
//   trainingFatPercentage: 0,
//   fatMethod: 'percentage',
//   // fatMethod: 'grams',
//   trainingFatGrams: 0,
//   restFatPercentage: 0
// });

const formValues = fromJS({
  trainingDay: 0,
  bmrCalculationMethod: 'harris-benedict',
  protein: 2,
  restDay: 0,
  calorieSplit: 'custom',
  activity: 1.2,
  restFatGrams: 0,
  bodyFat: 15.3,
  trainingFatPercentage: 0,
  fatMethod: 'percentage',
  // fatMethod: 'grams',
  trainingFatGrams: 0,
  restFatPercentage: 0
});

const state = Map().withMutations(map =>
  map
    .setIn(['userDetails', 'latestMeasurements'], fromJS(latestMeasurements))
    .setIn(['userDetails', 'dob'], moment('22-05-1988', 'DD-MM-YYYY').valueOf())
    .setIn(['userDetails', 'sex'], 'male')
    .setIn(['form', 'calorie-target', 'values'], formValues)
);

describe('calorie target props selector', () => {
  describe('bodyFat', () => {
    it('defaultValue => 15.3', () => {
      expect(selector(state).get('bodyFat')).toMatchSnapshot();
    });

    it('female, modified weight && sex => 12.6', () => {
      const modifiedState = state.withMutations(map =>
        map
          .setIn(['userDetails', 'latestMeasurements', 'weight'], 54)
          .setIn(['userDetails', 'sex'], 'female')
      );

      expect(selector(modifiedState).get('bodyFat')).toMatchSnapshot();
    });

    it('female, modified weight && sex && waist => 21.8 ', () => {
      const modifiedState = state.withMutations(map =>
        map
          .setIn(['userDetails', 'latestMeasurements', 'weight'], 61)
          .setIn(['userDetails', 'sex'], 'female')
          .setIn(['userDetails', 'latestMeasurements', 'waist'], 76)
      );

      expect(selector(modifiedState).get('bodyFat')).toMatchSnapshot();
    });
  });

  //minCalorie
  describe('minCalorie', () => {
    it('basic minimum calorie % => 73', () => {
      expect(selector(state).get('minCalorie')).toMatchSnapshot();
    });

    it('reducing height should reduce |minCalorie| => -71', () => {
      const modifiedState = state.withMutations(map =>
        map.setIn(['userDetails', 'latestMeasurements', 'height'], 160)
      );

      expect(selector(modifiedState)).toMatchSnapshot();
    });
  });

});
