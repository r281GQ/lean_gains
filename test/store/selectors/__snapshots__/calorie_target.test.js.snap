// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calorie target props selector bodyFat defaultValue => 15.3 1`] = `15.3`;

exports[`calorie target props selector bodyFat female, modified weight && sex && waist => 21.8  1`] = `21.8`;

exports[`calorie target props selector bodyFat female, modified weight && sex => 12.6 1`] = `12.6`;

exports[`calorie target props selector minCalorie basic minimum calorie % => 73 1`] = `-73`;

exports[`calorie target props selector minCalorie reducing height should reduce |minCalorie| => -71 1`] = `
Immutable.Map {
  bodyFat: 18,
  minCalorie: -71,
  max: Immutable.Map {
    maxRestGram: 152,
    maxRestPercentage: 71.9,
    maxTrainingGram: 152,
    maxTrainingPercentage: 71.9,
  },
  finalValues: Immutable.Map {
    rest: Immutable.Map {
      calorie: 1903,
      carbohydrate: 343,
      fat: 0,
      protein: 134,
    },
    training: Immutable.Map {
      calorie: 1903,
      carbohydrate: 343,
      fat: 0,
      protein: 134,
    },
  },
}
`;
